tests/std/tests/P2286R8_text_formatting_container_adaptors/test.cpp

```diff
@@ -0,0 +1,1206 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// derived from libc++'s test files:
+// * support/format.functions.common.h
+// * std/containers/container.adaptors/container.adaptors.format/format.functions.tests.h
+// * std/containers/container.adaptors/container.adaptors.format/format.functions.format.pass.cpp
+// * std/containers/container.adaptors/container.adaptors.format/format.functions.vformat.pass.cpp
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <charconv>
+#include <concepts>
+#include <format>
+#include <list>

```

Comment:
`<list>` is unused. Conversely, we should directly include the following:

```cpp
#include <cctype> // toupper
#include <cstddef> // size_t
#include <cstdint> // uint16_t
#include <cstring> // strlen
#include <functional> // greater
#include <iterator> // back_inserter
#include <locale> // locale
#include <string_view> // basic_string_view
#include <string> // basic_string
#include <type_traits> // type_identity_t
#include <utility> // forward
#include <vector> // vector
```

Some are included by `<test_format_support.hpp>`, but having tests directly include what they use makes it easier to refactor support headers.